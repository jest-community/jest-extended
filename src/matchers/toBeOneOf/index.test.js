import matcher from './';

expect.extend(matcher);

describe('.toBeOneOf', () => {
  it('passes when value is in given array', () => {
    expect(1).toBeOneOf([1, 2, 3]);
  });

  it('fails when value is not in given array', () => {
    expect(() => expect(4).toBeOneOf([1, 2, 3])).toThrowErrorMatchingSnapshot();
  });
});

describe('.not.toBeOneOf', () => {
  it('passes when value is not in given array', () => {
    expect(4).not.toBeOneOf([1, 2, 3]);
  });

  it('fails when value is in given array', () => {
    expect(() => expect(1).not.toBeOneOf([1, 2, 3])).toThrowErrorMatchingSnapshot();
  });
});
