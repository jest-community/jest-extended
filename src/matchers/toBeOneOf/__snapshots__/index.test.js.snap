// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toBeOneOf fails when value is in given array 1`] = `
"<dim>expect(</><red>item</><dim>).not.toBeOneOf(</><green>list</><dim>)</>

Expected value to not be in list:
  <green>[1, 2, 3]</>
Received:
  <red>1</>"
`;

exports[`.toBeOneOf fails when value is not in given array 1`] = `
"<dim>expect(</><red>item</><dim>).toBeOneOf(</><green>list</><dim>)</>

Expected value to be in list:
  <green>[1, 2, 3]</>
Received:
  <red>4</>"
`;
